argoverse/evaluation/stereo/eval.py

```diff
@@ -0,0 +1,348 @@
+
+    for abs_error_threshold, rel_error_threshold in zip(abs_error_thresholds, rel_error_thresholds):
+        bad_pixels = (abs_err > abs_error_threshold) & (rel_err > rel_error_threshold)
+
+        errors[f"num_errors_bg:{str(abs_error_threshold)}"] = np.sum(bg_mask & bad_pixels)
+        errors[f"num_errors_fg:{str(abs_error_threshold)}"] = np.sum(fg_mask & bad_pixels)
+        errors[f"num_errors_bg_est:{str(abs_error_threshold)}"] = np.sum(bg_mask & pred_mask & bad_pixels)
+        errors[f"num_errors_fg_est:{str(abs_error_threshold)}"] = np.sum(fg_mask & pred_mask & bad_pixels)
+
+    if save_disparity_error_image:
+        err = np.minimum(abs_err / abs_error_thresholds[0], rel_err / rel_error_thresholds[0])
+        disparity_error_image = np.zeros((*pred_disparity.shape, 3), dtype=np.uint8)
+
+        for threshold, color in log_colormap:
+            disparity_error_image[np.logical_and(err >= threshold[0], err < threshold[1])] = color
+
+        disparity_error_image[gt_disparity == 0] *= 0
+
+        disparity_error_image = np.uint8(
+            cv2.dilate(disparity_error_image, kernel=np.ones((2, 2), np.uint8), iterations=3)
+        )
+
+        # Get colormap image
+        fig = plt.figure(figsize=(24, 2))
+        ax = fig.add_subplot(111)
+        x = np.linspace(1, 11, 10)
+        y = np.linspace(0, 0, 10)
+        sizes = np.logspace(8, 12, num=10, base=2.0)
+        scalars = np.array(log_colormap, dtype=object)[:, 0] * abs_error_thresholds[0]
+        scalars = [f"[{scalar[0]:.2f}, {scalar[1]:.2f}]" for scalar in scalars]
+        colors = np.array(log_colormap, dtype=object)[:, 1]
+        colors = [color / 255.0 for color in colors]
+        plt.scatter(x, y, c=colors, s=3000)
+        plt.xlabel("Disparity range errors in pixels", fontsize=24)
+        plt.xticks(x, scalars, fontsize=20)
+        plt.yticks([])
+        ax.spines["top"].set_visible(False)
+        ax.spines["bottom"].set_visible(False)
+        ax.spines["right"].set_visible(False)
+        ax.spines["left"].set_visible(False)
+        plt.tight_layout()
+        plt.show()
+        fig.canvas.draw()
+        colormap_image = np.frombuffer(fig.canvas.tostring_rgb(), dtype=np.uint8)
+        colormap_image = colormap_image.reshape(fig.canvas.get_width_height()[::-1] + (3,))
+        plt.close()
+
+        # Add colormap to the disparity error image
+        c_img = colormap_image[:, :, ::-1]
+        d_img = disparity_error_image[:, :, ::-1]
+        x_offset = 30
+        y_offset = 30
+        d_img[y_offset : y_offset + c_img.shape[0], x_offset : x_offset + c_img.shape[1]] = c_img
+
+        log_id = str(gt_fpath).split("/")[-3]
+        timestamp = str(gt_fpath).split("_")[-1][:-4]
+        save_dir = f"{figs_fpath}/{log_id}/"
+        Path(save_dir).mkdir(parents=True, exist_ok=True)
+        cv2.imwrite(f"{save_dir}/disparity_error_{timestamp}.png", d_img)
+
+    return errors
+
+
+def accumulator(abs_error_thresholds: List) -> pd.DataFrame:
```

Comment:
I might prefer to name this `accumulate_stereo_metrics()` or something since `accumulator` seems like a class name.

Reply:
Agreed. Already changed.